### ISSUE TITLE
Provide a way to ban several users at once

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-server#9814]
In various scenarios we need to ban multiple users at once. Opening every single userprofile page, scrolling down and clicking "Ban", only works for very small amounts of profiles.

I'm happy to take (almost) whatever is easiest to implement:

* An API that allows passing a user id
* AN API that allows passing multiple user ids
* A web page where I can paste multiple user ids in a text area (one per line) and hit a "Ban all" button.

I am also open to ideas.

**Comment by wagnerand** [comment]
Setting p2 as this keeps coming up and there are currently several hundreds of accounts I need to ban.

**Comment by AlexandraMoga** [comment]
@wagnerand the Admin Users page, where all users are listed, gives you the option to select multiple users and then you can opt for "Ban selected users" from the drop down at the top of the list. Wouldn't this option help your usecase?

![image](https://user-images.githubusercontent.com/31961530/47550406-c6e9a700-d907-11e8-9a00-22480ab834b2.png)

**Comment by wagnerand** [comment]
Unfortunately, this does not work for me since the user ids that need to be deleted are not sequential and the search box doesn't allow searching for more than one id at a time.

**Comment by diox** [comment]
> Unfortunately, this does not work for me since the user ids that need to be deleted are not sequential and the search box doesn't allow searching for more than one id at a time.

This is probably how we should implement this, btw. Provide a way to search the admin user list using a list of ids. I've looked into this, and I think it's doable by overriding [ModelAdmin.get_search_results](https://docs.djangoproject.com/en/1.11/ref/contrib/admin/#django.contrib.admin.ModelAdmin.get_search_results), and do something different if commas are detected or if the search terms are all ids for instance, and fall back to the parent implementation otherwise. 

Alternatively... technically, the default implementation *does* support multiple search terms, it just considers that they should all be present. We could try adding a checkbox to the template and if present, in `get_search_result()` have it behave differently, making the queryset `OR` the multiple search terms instead of `AND`.

**Comment by diox** [comment]
@wagnerand would you have time to write tests (and check that the implementation works for you) and submit a PR for this using the following code in `UserAdmin`:

```python
    def get_search_results(self, request, queryset, search_term):
        """
        Returns a tuple containing a queryset to implement the search,
        and a boolean indicating if the results may contain duplicates.

        Copied from Django's, with a small difference: the operator joining the
        query parts is dynamic: if the search term contain a comma and no
        space, then the comma is used as the separator instead, and the query
        parts are joined by OR, not AND, allowing admins to search by a list of
        ids, emails or usernames and find all objects in that list.
        """
        # Apply keyword searches.
        def construct_search(field_name):
            if field_name.startswith('^'):
                return "%s__istartswith" % field_name[1:]
            elif field_name.startswith('='):
                return "%s__iexact" % field_name[1:]
            elif field_name.startswith('@'):
                return "%s__search" % field_name[1:]
            else:
                return "%s__icontains" % field_name

        use_distinct = False
        search_fields = self.get_search_fields(request)
        filters = []
        joining_operator = operator.and_
        if search_fields and search_term:
            orm_lookups = [construct_search(str(search_field))
                           for search_field in search_fields]
            if ' ' not in search_term and ',' in search_term:
                separator = ','
                joining_operator = operator.or_
            else:
                separator = None
            for bit in search_term.split(separator):
                or_queries = [models.Q(**{orm_lookup: bit})
                              for orm_lookup in orm_lookups]

                q_for_this_term = models.Q(reduce(operator.or_, or_queries))
                filters.append(q_for_this_term)

            if not use_distinct:
                for search_spec in orm_lookups:
                    if admin.utils.lookup_needs_distinct(
                            self.opts, search_spec):
                        use_distinct = True
                        break

        if filters:
            queryset = queryset.filter(reduce(joining_operator, filters))
        return queryset, use_distinct
```

(See the docstring for how it works)

**Comment by wagnerand** [comment]
Nice! Thanks @diox!

**Comment by AlexandraMoga** [comment]
Verified on AMO -dev with FF63, Win10x64
I've managed to pull out multiple users when searching by IDs - comma separated and no spacing. Results were returned when I searched for 40 Ids, but at 50 Ids I've received a gateway timeout error (504).I guess this is an acceptable limit. 
I also think that the actual banning process will take a while and we might run into timeout errors.
Maybe prod will behave better, being more stable overall. 

![image](https://user-images.githubusercontent.com/31961530/47721165-32a97800-dc58-11e8-9a75-c43bf02a4eba.png)
